docs/realm.js

```diff
@@ -499,6 +499,33 @@ class Realm {
  *   name. Queries can be done using both the public and the underlying property name.
  */
 
+/**
+ * The retreived schema from an open Realm has a different structure as the input schema.
+ * @typedef Realm~CanonicalObjectSchema
+ * @type {Object}
+ * @property {string} name - Represents the object type.
+ * @property {string} [primaryKey] - The name of a `"string"` or `"int"` property
+ *   that must be unique across all objects of this type within the same Realm.
+ * @property {boolean} [embedded] - True if the object type is embedded. An embedded object
+ *   can be linked to by at most one parent object. Default value: false.
```

Comment:
🔜 I think if it would be better to make the docs model the Typescript schema instead of duplicating content, i.e. have the base class and the classes which extend it in the docs as per [SubscriptionSet](https://github.com/realm/realm-js/blob/master/docs/flexible-sync.js#L308) for example – JSDoc renders this sensibly by showing all the base properties on the class in question: https://www.mongodb.com/docs/realm-sdks/js/latest/Realm.App.Sync.SubscriptionSet.html

Reply:
mmm...makes sense I guess.  I'll have to see if I can get that figured out.  SubscrionSet is a class and this is a pojo